modules/service/src/main/scala/lucuma/odb/service/CalibrationsService.scala

```diff
@@ -145,14 +146,22 @@ object CalibrationsService extends CalibrationObservations {
           }
         } else none.pure[F]
 
-      private def allObservations(pid: Program.Id, selection: ObservationSelection)(using Transaction[F]): F[(List[(Observation.Id, Config.GmosNorth | Config.GmosSouth)])] =
+      private def allValidObservations(obs: Map[Observation.Id, EitherNel[GeneratorParamsService.Error, GeneratorParams]]): List[(Observation.Id, Config.GmosNorth | Config.GmosSouth)] =
+        obs.toList.collect {
+          case (oid, Right(GeneratorParams(Right(_), mode: Config.GmosNorth, _))) => (oid, mode)
+          case (oid, Right(GeneratorParams(Right(_), mode: Config.GmosSouth, _))) => (oid, mode)
+        }
+
+      private def allCalibObservations(obs: Map[Observation.Id, EitherNel[GeneratorParamsService.Error, GeneratorParams]]): List[(Observation.Id, Config.GmosNorth | Config.GmosSouth)] =
+        obs.toList.collect {
+          case (oid, Right(GeneratorParams(_, mode: Config.GmosNorth, _))) => (oid, mode)
+          case (oid, Right(GeneratorParams(_, mode: Config.GmosSouth, _))) => (oid, mode)
+        }
```

Comment:
`allCalibObservations` is misleading, I think, since it doesn't filter for calibration observations.  It will pick up twilight targets though as well as everything else in the map that it sent to it.

Reply:
Right, we feed this a query with only calibrations and we are including both with and without itc, that's why it is `all`